module_vd_gas_wesely.f90

```diff
@@ -0,0 +1,211 @@
+                rs = amax1(rs,rmin)
+            endif
+            goto 100 ! need to remove goto blocks - they are not accepted by NCO
+        endif 
+
+        !!Step 3-2. normal case, surface layer over land 
+        !!!Step 3-2-1.stomatal and mesophyll resistance,Rst & Rm
+        rst = rmax
+        if (ts>0 .and. ts<40) then
+            rst = diffrat*ri*(1.+(200./(solflux + 0.1))**2)*(400./(ts*(40.-ts)))  !Follow Wesely et al.,(1989) eqn (3)&(4)   
+            rst = rst * dstress(istress)
+        endif
+        !Effect of dew and rain,increased by a factor of 3, as leafs are covered
+        if (iwet >0) then
+            rst = 3.* rst                                                                         
+        endif
+        
+        rst = amin1(rmax,rst)
+
+        !Follow Wesely et al., (1989) eqn (6)
+        rm = 1./(henry/3000.+100.*f0)                                                             
+        rm = amax1(rmin,rm)
+        rm = amin1(rmax,rm)
+
+    !   !!Step 3-2-2.upper canopy resistance,Ruc
+        if (iwet == 0) then                     !dry surface
+            ruc = rlu/(henry/henso2+f0)
+            ruc = amin1(rmax,ruc)
+        else
+            if (iwet == 1) then                 !dew surface
+                rlus = 100.
+                if (ilu == 1) then
+                    rlus = 50.
+                endif
+    !            rluo = 1./(1./3000.+1./(3.*rlu))                                                   !Follow Wesely et al.,(1989) eqn (11)
+                rluo = 1000 + rlu
+            else                                !rain surface
+    !            rlus = 1./(1./5000. + 1./(3.*rlu))                                                 !Follow Wesely et al.,(1989) eqn (12)
+                rlus = 2000.+rlu
+                if (ilu == 1.) then
+                    rlus = 50.
+                endif
+                rluo = 1./(1./1000. + 1./(3.*rlu))                                                 !Follow Wesely et al.,(1989) eqn (13)
+            endif
+            
+            if (iso2 == 1) then
+                ruc = rlus
+            elseif (io3 == 1) then
+                ruc = rluo
+            else
+    !            ruc = 1./(1./(3.*rlu) + 1.e-7*henry + f0/rluo)                                     !Follow Wesely et al.,(1989) eqn (14)
+                ruc = 1./(henry/(henso2*rlus) + f0/rluo)
+            endif
+        endif
+
+        !!!Step 3-2-3.buyant convection and lower canopy resistance, Rdc & Rlc
+        rdc = 100.*(1. +1000./(solflux +10.))                  !Follow Wesely (1989) eqn (5),but set terrain slope factor to be 1. 
+        rdc = amin1(rmax, rdd)
+
+        rlc = 1./(henry/(henso2*rlcs)+f0/rlco)                 !Follow Wesely et al. (1989) eqn (8)
+        rlc = amin1(rmax,rlc)
+
+        !!!Step 3-2-4.ground surface resistance, Rgs
+        !!!Rac was read-in ???
```

Comment:
rac is read in?

Reply:
see line 42.  You have it listed as an input variable